### ISSUE TITLE
Disabling submit button

### ISSUE THREAD

**Issue by augustskare** [edmundhung/conform#73]
If I disable the submit button on submission it looks like the `state` of `useForm` is "lagging" behind. Clicking the submit button a second time will make it update (and you'll see the error in the example below). 

My implementation is in Remix, not sure if that's related. Here's a reproduced example on [CodeSandbox](https://codesandbox.io/p/sandbox/suspicious-wildflower-3sdopv?file=/app/routes/index.tsx)

**Comment by edmundhung** [comment]
Hi @augustskare. Thanks for the bug report. Your problem should be fixed in [v0.5.0-pre.0](https://github.com/edmundhung/conform/releases/tag/v0.5.0-pre.0) as there is an improvement on how form level error is populated.

If you wanna know more, here is what happened:

Conform utilise the [element.setCustomValidity()](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setCustomValidity) API to report an error by mapping the name on the input with the message. Before v0.5, Conform will just assigns form level error to the first form element with name = "", which usually end up assigned to your submit button as `element.name` returns empty string even when no name attribute is set on the button.   

However, similar to the FormData API, the setCustomValidity API will be silently ignored if it is disabled. This is where it goes wrong in your case: It tries to report the form-level error while your button is still disabled.

When a form is submitted, Remix transition state goes from `idle` to `submitting` (calling the action), then move to `loading` for revalidation (calling the loader) and back to `idle` when done. The actionData is changed before the revalidation is complete and Conform is trying to report the error while the `state` is still `loading` .

Personally, I will just disable the fields with `state === 'submitting'`. However, I think there are also valid concern to use `state !== 'idle'`. If checking for idle fit your needs better, you can delay the actionData changes before v0.5 is ready:

```ts
const form = useForm({
  // Update the state only when the fields is not disabled
  // There is no need to do this if you are disabling only when state === 'submitting' 
  state: transition.state === 'idle' ? state : undefined
});
```

> Note: The solution above is also useful if you want to disable the inputs on form submission, which v0.5 will not solves the problem for you.

**Comment by augustskare** [comment]
Thank you for the detailed response! 😍 

Checking the transition state on the `state` property solves my issue 😊
